### PR TITLE
Basic fix for #652

### LINKED ISSUE #652: git push to branch not of same name: use git's suggestion
I checked out a branch with a different name than the remote counterpart:

```
$ git checkout -b local-branch myupstream/remote-branch
```

When I attempted to push my changes upstream, I got the following error:
```
$ git push -f
fatal: The upstream branch of your current branch does not match
the name of your current branch.  To push to the upstream branch
on the remote, use

    git push myupstream HEAD:remote-branch

To push to the branch of the same name on the remote, use

    git push myupstream local-branch
```

However, thefuck didn't use this information to provide a suggestion:
```
$ fuck
git push -d [enter/↑/↓/ctrl+c] 
```

### PR DESCRIPTION
**Pull request by Epse**
This will probably fix #652

### CODE REVIEW

File: thefuck/rules/git_push_different_branch_names.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+@git_support
+def match(command):
+    return "git push" in command.script and "The upstream branch of your current branch does not match" in command.output
+
+
+@git_support
+def _parse_git_output(output):
+    regex = re.compile(r'^ +(git push [^\s]+ [^\s]+)', re.MULTILINE)
+    return regex.findall(output)[0]
```
Comment (scorphus) [inline thefuck/rules/git_push_different_branch_names.py]: These 2 lines can be replaced by one:

```python
return re.findall(r'^ +(git push [^\s]+ [^\s]+)', command.output, re.MULTILINE)
```

And maybe move it inside `get_new_command`, while **fixing** the latter too.

**Comment by Epse** [comment]
oops yeah didn't push everything here

**Comment by Epse** [comment]
@scorphus is this one ok now?

**Comment by Epse** [comment]
I must have had a retarded brainfart writing the test on this one. Really. Thing is fine now. Sorry for the delay.

File: thefuck/rules/git_push_different_branch_names.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import re
+from thefuck.specific.git import git_support
+
+
+@git_support
+def match(command):
+    return "git push" in command.script and "The upstream branch of your current branch does not match" in command.output
```
Comment (scorphus) [inline thefuck/rules/git_push_different_branch_names.py]: `@git_support` already asserts the presence of `git` in script.

File: thefuck/rules/git_push_different_branch_names.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+
+@git_support
+def get_new_command(command):
+    return re.findall(r'^ +(git push [^\s]+ [^\s]+)', command.output, re.MULTILINE)[0]
+
+
+enabled_by_default = True
```
Comment (scorphus) [inline thefuck/rules/git_push_different_branch_names.py]: Rules are enabled by default.

File: README.md [hunk 4]
```diff
@@ -196,6 +196,7 @@ using the matched rule and runs it. Rules enabled by default are as follows:
 * `git_pull_clone` &ndash; clones instead of pulling when the repo does not exist;
 * `git_pull_uncommitted_changes` &ndash; stashes changes before pulling and pops them afterwards;
 * `git_push` &ndash; adds `--set-upstream origin $branch` to previous failed `git push`;
+* `git_push_different_branch_names` &ndash; fixes pushes when local brach name does not match remote branch name ;
```
[the remark this answers is not in the document]
Comment (Epse) [reply README.md]: Auwch. How did I not spot that?

**Comment by Epse** [comment]
should be all fixed now

**Comment by nvbn** [comment]
@scorphus, thanks!

**Comment by scorphus** [comment]
Thanks for another contribution, @Epse! 👍 👊

**Pull request merged**